Classes/Common/MetsDocument.php

```diff
@@ -589,70 +590,6 @@ class_exists($class)
                 $metadata['title_sorting'][0] = '';
             }
-            // Add collections and owner from database to toplevel element if document is already saved.
-            if (
-                \TYPO3\CMS\Core\Utility\MathUtility::canBeInterpretedAsInteger($this->uid)
-                && $id == $this->_getToplevelId()
-            ) {
-                $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
-                    ->getQueryBuilderForTable('tx_dlf_documents');
-
-                $result = $queryBuilder
-                    ->select(
-                        'tx_dlf_collections_join.index_name AS index_name'
-                    )
-                    ->from('tx_dlf_documents')
-                    ->innerJoin(
-                        'tx_dlf_documents',
-                        'tx_dlf_relations',
-                        'tx_dlf_relations_joins',
-                        $queryBuilder->expr()->eq(
-                            'tx_dlf_relations_joins.uid_local',
-                            'tx_dlf_documents.uid'
-                        )
-                    )
-                    ->innerJoin(
-                        'tx_dlf_relations_joins',
-                        'tx_dlf_collections',
-                        'tx_dlf_collections_join',
-                        $queryBuilder->expr()->eq(
-                            'tx_dlf_relations_joins.uid_foreign',
-                            'tx_dlf_collections_join.uid'
-                        )
-                    )
-                    ->where(
-                        $queryBuilder->expr()->eq('tx_dlf_documents.pid', intval($cPid)),
-                        $queryBuilder->expr()->eq('tx_dlf_documents.uid', intval($this->uid))
-                    )
-                    ->orderBy('tx_dlf_collections_join.index_name', 'ASC')
-                    ->execute();
-
-                $allResults = $result->fetchAll();
-
-                foreach ($allResults as $resArray) {
-                    if (!in_array($resArray['index_name'], $metadata['collection'])) {
-                        $metadata['collection'][] = $resArray['index_name'];
-                    }
-                }
-
-                $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
-                    ->getQueryBuilderForTable('tx_dlf_documents');
-
-                $result = $queryBuilder
-                    ->select(
-                        'tx_dlf_documents.owner AS owner'
-                    )
-                    ->from('tx_dlf_documents')
-                    ->where(
-                        $queryBuilder->expr()->eq('tx_dlf_documents.pid', intval($cPid)),
-                        $queryBuilder->expr()->eq('tx_dlf_documents.uid', intval($this->uid))
-                    )
-                    ->execute();
-
-                $resArray = $result->fetch();
-
-                $metadata['owner'][0] = $resArray['owner'];
-            }
```

Comment:
Why remove this instead of adding checks as in #752? (Though I guess the DB queries should be extracted to DocumentRepository.)

Reply:
Actually, we have no uid at this point anymore. The `Doc::class` should only keep the information from the METS-file. Everything else should come from `Document::class` which has access to the `DocumentRepository`. 

So we can't execute the queries at this point anyhow and we can't even use the documentRepository. And I think it's not necessary at all.